data/model/tag.py

```diff
@@ -396,20 +397,27 @@ def create_temporary_hidden_tag(repo, image_obj, expiration_s):
     """
     Create a tag with a defined timeline, that will not appear in the UI or CLI.
 
-    Returns the name of the temporary tag.
+    Returns the name of the temporary tag or None on error.
     """
     now_ts = get_epoch_timestamp()
     expire_ts = now_ts + expiration_s
     tag_name = str(uuid4())
-    RepositoryTag.create(
-        repository=repo,
-        image=image_obj,
-        name=tag_name,
-        lifetime_start_ts=now_ts,
-        lifetime_end_ts=expire_ts,
-        hidden=True,
-    )
-    return tag_name
+
+    # Ensure the repository is not marked for deletion.
+    with db_transaction():
+        current = Repository.get(id=repo)
+        if current.state == RepositoryState.MARKED_FOR_DELETION:
```

Comment:
Do you think this will heavily impact the database load?

Reply:
I don't believe so, as we already create this under a transaction elsewhere and it should only be per uploaded blob. That being said, it will have a small impact, but its the only safe way.